docs/source/reference/udfs/custom.rst

```diff
@@ -0,0 +1,142 @@
+
+Setup
+-----
+
+An abstract method that must be implemented in your child class. The `setup` function can be used to initialize the parameters for executing the UDF. The following parameters must be set:
+
+- `cacheable`: bool
+
+  - `True`: Cache should be enabled. The cache will be automatically invalidated when the UDF changes.
+  - `False`: Cache should not be enabled.
+
+- `udf_type`: str
+
+  - `object_detection`: UDFs for object detection.
+
+- `batchable`: bool
+
+  - `True`: Batching should be enabled.
+  - `False`: Batching is disabled.
+
+The custom setup operations for the UDF can be written inside the function in the child class. If no custom logic is required, then you can just write `pass` in the function definition.
+
+Example of the `setup` function:
+
+.. code-block:: python
+
+    @setup(cachable=True, udf_type="object_detection", batchable=True)
+    def setup(self, threshold=0.85):
+        self.threshold = threshold
+        self.model = torchvision.models.detection.fasterrcnn_resnet50_fpn(
+            weights="COCO_V1", progress=False
+        )
+        self.model.eval()
+
+Forward
+--------
+
+An abstract method that must be implemented in your child class. The `forward` function receives the frames and runs the Deep Learning model on the frames. The logic for transforming the frames and running the models must be provided by you. The arguments that need to be passed are:
+
+- `input_signatures`: List[IOColumnArgument]
+
+  Data types of the inputs to the `forward` function must be specified. If no constraints are given, then no validation is done for the inputs.
+
+- `output_signatures`: List[IOColumnArgument]
+
+  Data types of the outputs to the `forward` function must be specified. If no constraints are given, then no validation is done for the inputs.
+
+A sample `forward` function is given below:
+
+.. code-block:: python
+    
+    @forward(
+        input_signatures=[
+            PyTorchTensor(
+                name="input_col",
+                is_nullable=False,
+                type=NdArrayType.FLOAT32,
+                dimensions=(1, 3, 540, 960),
+            )
+        ],
+        output_signatures=[
+            PandasDataframe(
+                columns=["labels", "bboxes", "scores"],
+                column_types=[
+                    NdArrayType.STR,
+                    NdArrayType.FLOAT32,
+                    NdArrayType.FLOAT32,
+                ],
+                column_shapes=[(None,), (None,), (None,)],
+            )
+        ],
+    )
+    def forward(self, frames: Tensor) -> pd.DataFrame:
+        predictions = self.model(frames)
+        outcome = []
+        for prediction in predictions:
+            pred_class = [
+                str(self.labels[i]) for i in list(self.as_numpy(prediction["labels"]))
+            ]
+            pred_boxes = [
+                [i[0], i[1], i[2], i[3]]
+                for i in list(self.as_numpy(prediction["boxes"]))
+            ]
+
+
```

Comment:
can we give an explanation of the example like the one for setup? 
The input to the forward function is a PyTorch tensor of shape (1, 3, 540, 960) and type Float32. The output is a pandas dataframe that has 3 columns with names "labels", "bboxes" and "scores" and type string, float32 and float32 respectively.